server/service.py

```diff
@@ -42,11 +43,11 @@ def add_transaction(payload):
         payload["tax"] = 0
     if payload["tip"] is None:
         payload["tip"] = 0
-    if payload["ip"] is None:
+    if "ip" not in payload or payload["ip"] is None:
         payload["ip"] = ""
-
-    # Generate transaction ID
-    transaction_id = str(uuid.uuid4())
+    if "id" not in payload:
+        # If id is not present in the payload, generate one.
```

Comment:
When could this happen? Should we log something in this case?

Reply:
For backward compatibility, we will not fail if an ID is not supplied, e.g. a user has an old version of the app.